### DISCUSSION TITLE
[FEATURE] PnP commands to publish a SPO site template to the tenant

### DISCUSSION THREAD

**Discussion by rachoksh** [pnp/powershell#3069] [General]
**Is your feature request related to a problem? Please describe.**
Problem is currently there is no way to publish a custom SPO modern template to the tenant.

I am looking for a way to use the .pnp package from the below command and use that to publish a custom SPO site template to the Tenant that appears under the "From your Organization" in the UI for end users to be able to choose an Organization template and apply. This would include lists, libraries, navigation, site logo as well as client-side custom pages.

Get-PnPSiteTemplate -out template.pnp -IncludeAllClientSidePages -PersistBrandingFiles

**Describe the solution you'd like**
Solution could be something like below two commands - 

Import-PnPTenantTemplate -Path ./template.pnp
&
Publish-PnPTenantTemplate -Identity "Template Name" -Scope Organization

**Describe alternatives you've considered**
Current solution is for Admins to use this - Invoke-PnPSiteTemplate .\template.pnp.
End users cannot do this as they don't know PnP PS.

Publishing to the tenant allows end users to pick from any of the published organization templates in SPO.

**Comment by veronicageek** [comment]
@rachoksh - Are Site scripts & Site designs not what you're after?

**Comment by gautamdsheth** [comment]
@rachoksh - what you are looking is site templates and scripts.

You can read more about it here:

https://learn.microsoft.com/en-us/sharepoint/dev/declarative-customization/get-started-create-site-design

You can call PnP Provisioning from the site scripts. The provisioning needs to be run on a separate hosted environment or server. You can consider  Az function or runbooks or something else.
